cassandra/src/main/java/org/hibernate/ogm/datastore/cassandra/CassandraDialect.java

```diff
@@ -183,14 +200,23 @@ public void insertOrUpdateTuple(EntityKey key, Tuple tuple, TupleContext tupleCo
 
 			// insert and update are both 'upsert' in cassandra.
 			Insert insert = insertInto( quote( key.getTable() ) );
-			Object[] columnValues = new Object[updateOps.size()];
+			List<Object> columnValues = new LinkedList<>();
+			Set<String> seenColNames = new HashSet<>();
 			for ( int i = 0; i < updateOps.size(); i++ ) {
 				TupleOperation op = updateOps.get( i );
 				insert.value( quote( op.getColumn() ), QueryBuilder.bindMarker() );
-				columnValues[i] = op.getValue();
+				columnValues.add( op.getValue() );
+				seenColNames.add( op.getColumn() );
+			}
+			for ( int j = 0; j < key.getColumnNames().length; j++ ) {
```

Comment:
@jhalliday, e.g. that's one of the failing tests: `org.hibernate.ogm.backendtck.associations.manytoone.ManyToOneTest.testBidirectionalManyToOneRegular()`.

The "DDL" is:

    CREATE TABLE IF NOT EXISTS "SalesForce" ("id" text, "corporation" text, PRIMARY KEY ("id"));
    CREATE TABLE IF NOT EXISTS "SalesGuy" ("id" text, "name" text, "salesForce_id" text, PRIMARY KEY ("id"));
    CREATE INDEX IF NOT EXISTS "SalesGuy_salesForce_id" ON "SalesGuy" ("salesForce_id")

The failing statement is

    SELECT * FROM "SalesGuy" WHERE "salesForce_id"=? ALLOW FILTERING;

With the param `[sales_force]`.  Exception is:

    com.datastax.driver.core.exceptions.InvalidQueryException: No indexed columns present in by-columns clause with Equal operator